Db/system both rebase

derekwbrown opened the pull request:
### What does this PR do?

Implements the IO/disk check for windows.  This version uses WMI, to match agent5.  Have an eventual PR that uses PDH.

Moves platform specific code to conditionally compiled rather than runtime evaluation.

Note that this does NOT detect insertion of drives, at least as tested.  Is same behavior as agent5.

(Tested using virtualbox, and using Disk Manager to mount & unmount drives).

truthbk committed:
[load] adding system check.

truthbk committed:
[load] adding system check.

truthbk committed:
[system][iostats] adding iostats check + tests.

truthbk committed:
[system][iostats] adding iostats check + tests.

truthbk committed:
[iostats] more precise comment..

truthbk committed:
[iostats] more precise comment..

truthbk committed:
[uptime] adding check + tests.

truthbk committed:
[load] removing logging statement.

truthbk committed:
[load] removing logging statement.

truthbk committed:
[iostats][test] amending expected number of asserted calls.

truthbk committed:
[iostats] improve windows support.

truthbk committed:
[iostats] improve windows support.

truthbk committed:
[iostats] refactor unix specific io submission, for clarity.

truthbk committed:
[iostats] refactor unix specific io submission, for clarity.

truthbk committed:
[iostats] refactor, windows numbers differ greatly. Currently not BW compatible.

truthbk committed:
[iostats] refactor, windows numbers differ greatly. Currently not BW compatible.

truthbk committed:
[iostats] removing unnecessary C(go) header.

truthbk committed:
[iostats] removing unnecessary C(go) header.

truthbk committed:
[iostats] implement device blacklist.

truthbk committed:
[iostats] implement device blacklist.

truthbk committed:
[iostats] fixing blacklist management + adding test.

truthbk committed:
[iostats] fixing blacklist management + adding test.

derekwbrown committed:
[windows] implement windows IO check

derekwbrown committed:
[windows] implement windows IO check

derekwbrown committed:
Initial checkin of code using both WMI & PDH

derekwbrown committed:
Initial checkin of code using both WMI & PDH

derekwbrown committed:
Modified APM request

derekwbrown committed:
Modified APM request

derekwbrown committed:
touch up logging.  Remove calls to APM

derekwbrown committed:
touch up logging.  Remove calls to APM

derekwbrown committed:
Switch back to wmi, at least for now.  Neither PDH nor WMI is picking up disk changes, at least in the manner tested.

derekwbrown committed:
Merge branch 'db/system-both' into db/system-both-rebase

derekwbrown committed:
Fix problems with merge

derekwbrown committed:
Fix merge.   Currently uses WMI; will re-add pdh based IO check at a later
date

gmmeyer committed:
centos6 changes (#354)

* adds upstart file to centos6

* adds upstart file to rpm

* changes conditional

* enables alternative python

truthbk reviewed (commented):
This looks absolutely fine overall. Left a few comments/questions, small things that might make it a little more idiomatic or nitpicks. Also left a couple of notes to myself - as this PR is a bit of a joint effort..

derekwbrown committed:
Changes to reflect review feedback

derekwbrown committed:
Fix gitlab ci so ci testing can continue

truthbk reviewed (approved):
Logic looks good to me. There's a few things I commented on, but I'll let you decide if/when you want to address those.

derekwbrown committed:
Final review feedback.  Make string conversion more efficient by using a
byte buffer to do the original conversion, rather than appending a
character on to the string for each pass.

derekwbrown committed:
Final review feedback. Make string conversion more efficient by using a
byte buffer to do the original conversion, rather than appending a
character on to the string for each pass.
